src/js/models/documentexports.js

```diff
@@ -48,6 +49,35 @@ define(function (require, exports, module) {
         layerExportsMap: Immutable.Map()
     });
 
+    Object.defineProperties(DocumentExports.prototype, objUtil.cachedGetSpecs({
+        /**
+         * Produce an array of ExportAssets for the root of this document, converted to a JS object
+         * This is used for serializing the data to Ps metadata store
+         *
+         * @return {Array.<object>} An array of ExportAsset-like JS objects
+         */
+        rootExportsArray: function () {
+            if (this.rootExports && this.rootExports.size > 0) {
+                return this.rootExports.toJS();
```

Comment:
Again, I'm generally curious about the mixture of mutable and immutable data structures. I'm not saying that it's necessarily bad, but I wonder what the constraints were that lead to a mix of both. It's almost always _simpler_ if you only use mutable or immutable data structures, though of course there are valid reasons to use a mixture.